platform-tests/src/api/api_test.go

```diff
@@ -0,0 +1,130 @@
+package api
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"time"
+
+	"github.com/cloudfoundry-community/go-cfclient"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func lg(things ...interface{}) {
+	fmt.Fprintln(GinkgoWriter, things...)
+}
+
+var _ = Describe("API Availability Monitoring", func() {
+
+	It("should have uninterupted access to cloudfoundry api during deploy", func() {
+
+		cfg := &cfclient.Config{
+			ApiAddress:        fmt.Sprintf("https://api.%s", os.Getenv("SYSTEM_DNS_ZONE_NAME")),
+			Username:          os.Getenv("CF_USER"),
+			Password:          os.Getenv("CF_PASS"),
+			SkipSslValidation: os.Getenv("SKIP_SSL_VALIDATION") == "true",
+		}
+		Expect(cfg.ApiAddress).ToNot(BeEmpty(), "SYSTEM_DNS_ZONE_NAME environment variable must be set")
+		Expect(cfg.Username).ToNot(BeEmpty(), "CF_USER environment variable must be set")
+		Expect(cfg.Password).ToNot(BeEmpty(), "CF_PASS environment variable must be set")
+
+		deployment := &Deployment{
+			AtcAddress: os.Getenv("CONCOURSE_ATC_URL"),
+			Password:   os.Getenv("CONCOURSE_ATC_PASSWORD"),
+			Username:   os.Getenv("CONCOURSE_ATC_USERNAME"),
+			Version:    os.Getenv("PIPELINE_TRIGGER_VERSION"),
+			Team:       "main",
+		}
+		Expect(deployment.AtcAddress).ToNot(Equal(""), "CONCOURSE_ATC_URL environment variable must be set")
+		Expect(deployment.Password).ToNot(Equal(""), "CONCOURSE_ATC_PASSWORD environment variable must be set")
+		Expect(deployment.Username).ToNot(Equal(""), "CONCOURSE_ATC_USERNAME environment variable must be set")
+		Expect(deployment.Version).ToNot(Equal(""), "PIPELINE_TRIGGER_VERSION environment variable must be set")
+
+		monitor := NewMonitor(cfg)
+
+		monitor.Add("Listing all apps in a space", func(cf *cfclient.Client) error {
+			org, err := cf.GetOrgByName("admin")
+			if err != nil {
+				return fmt.Errorf("Failed to fetch 'admin' org: %s", err)
+			}
+
+			space, err := cf.GetSpaceByName("healthchecks", org.Guid)
+			if err != nil {
+				return fmt.Errorf("Failed to fetch 'healthchecks' space within 'admin' org: %s", err)
+			}
+			query := url.Values{}
+			query.Add("q", fmt.Sprintf("organization_guid:%s", org.Guid))
+			query.Add("q", fmt.Sprintf("space_guid:%s", space.Guid))
+			apps, err := cf.ListAppsByQuery(query)
+			if err != nil {
+				return fmt.Errorf("Failed to fetch apps within space 'healthchecks' in org 'admin': %s", err)
+			} else if len(apps) < 1 {
+				return fmt.Errorf("expected at least one app in the 'healthchecks' space got %d", len(apps))
+			}
+
+			return nil
+		})
+
+		monitor.Add("Fetching detailed app information", func(cf *cfclient.Client) error {
+			org, err := cf.GetOrgByName("admin")
+			if err != nil {
+				return fmt.Errorf("Failed to fetch 'admin' org")
+			}
+
+			apps, err := cf.ListAppsByQuery(url.Values{"q": []string{
```

Comment:
Interested again, you seem to have changed the style of how you build `url.Values`. Any reason for that?